src/sage/symbolic/relation.py

```
@@ -576,12 +576,105 @@ def string_to_list_of_solutions(s):
+        [x - 1, y - 2]
+        sage: _normalize_to_nonrelational(x - 1)
+        x - 1
+        sage: _normalize_to_nonrelational([x - 1, y - 2])
+        [x - 1, y - 2]
+        sage: _normalize_to_nonrelational((x - 1, y - 2))
+        [x - 1, y - 2]
+        sage: _normalize_to_nonrelational(1)
+        1
+        sage: _normalize_to_nonrelational(True)
+        0
+        sage: _normalize_to_nonrelational(False)
+        1
+    """
+    if isinstance(f, (list, tuple)):
+        return [_normalize_to_nonrelational(g) for g in f]
+    if isinstance(f, bool):
+        return 0 if f else 1
+    from sage.symbolic.expression import Expression
+    if isinstance(f, Expression) and f.is_relational():
+        assert f.operator() == operator.eq
+        return f.lhs() - f.rhs()
+    return f
+
+
+def _normalize_to_list_expressions(f) -> list:
+    """
+    Normalize either a single equation or a list of equations into a list of
+    equations.
+
+    TESTS::
+
+        sage: from sage.symbolic.relation import _normalize_to_list_expressions
+        sage: x, y = var('x, y')
+        sage: _normalize_to_list_expressions(x == 1)
+        [x == 1]
+        sage: _normalize_to_list_expressions([x == 1, y == 2])
+        [x == 1, y == 2]
+        sage: _normalize_to_list_expressions({})
+        Traceback (most recent call last):
+        ...
+        TypeError: must be a symbolic expression or a list of symbolic expressions
+    """
+    from sage.symbolic.expression import Expression
+    if isinstance(f, (Expression, bool)):
+        f = [f]
+    if not isinstance(f, (list, tuple)) or not all(isinstance(s, (Expression, bool)) for s in f):
+        raise TypeError("must be a symbolic expression or a list of symbolic expressions")
+    return f
+
+
 ###########
 # Solving #
 ###########
 
 
-def solve(f, *args, **kwds):
+def solve(f, *args, explicit_solutions=None, multiplicities=None, to_poly_solve=None,
+          solution_dict=False, algorithm=None, domain=None):
```

Another behavioral change: now extra keyword argument raise error instead of silently ignored